powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthAuthenticatorProvider.java

```
@@ -0,0 +1,354 @@
+            activationResponse.setDevicePublicKeyBase64(activation.getDevicePublicKeyBase64());
+
+            // Generate authenticator detail
+            return convert(activationResponse, applicationEntity);
+        } catch (GenericCryptoException ex) {
+            logger.error(ex.getMessage(), ex);
+            // Rollback is not required, cryptography errors can only occur before writing to database
+            throw new Fido2AuthenticationFailedException("Generic cryptography error");
+        } catch (CryptoProviderException ex) {
+            logger.error(ex.getMessage(), ex);
+            // Rollback is not required, cryptography errors can only occur before writing to database
+            throw new Fido2AuthenticationFailedException("Invalid cryptography provider");
+        } catch (JsonProcessingException e) {
+            throw new Fido2AuthenticationFailedException("Unable to serialize extras");
+        } catch (GenericServiceException e) {
+            throw new Fido2AuthenticationFailedException("Generic service exception");
+        }
+
+    }
+
+    private AuthenticatorDetail convert(Activation activation, ApplicationEntity application) {
+        final AuthenticatorDetail authenticatorDetail = new AuthenticatorDetail();
+
+        authenticatorDetail.setApplicationId(activation.getApplicationId());
+        authenticatorDetail.setUserId(activation.getUserId());
+        authenticatorDetail.setActivationId(activation.getActivationId());
+        authenticatorDetail.setActivationStatus(activation.getActivationStatus());
+        authenticatorDetail.setActivationName(activation.getActivationName());
+        authenticatorDetail.setExternalId(activation.getExternalId());
+        try {
+            authenticatorDetail.setExtras(objectMapper.readValue(activation.getExtras(), new TypeReference<HashMap<String,Object>>() {}));
+        } catch (JsonProcessingException e) {
+            //
```

❗ The exception is swallowed.

---

Already fixed in separate pull request.